pkg/builder/cluster/convert/convert.go

```diff
@@ -105,48 +105,76 @@ var (
 )
 
 // TODO(mattmoor): Should we move this somewhere common, because of the flag?
-func gitToContainer(git *v1alpha1.GitSourceSpec) (*corev1.Container, error) {
+func gitToContainer(git *v1alpha1.GitSourceSpec, name string) (*corev1.Container, error) {
 	if git.Url == "" {
 		return nil, validation.NewError("MissingUrl", "git sources are expected to specify a Url, got: %v", git)
 	}
 	if git.Revision == "" {
 		return nil, validation.NewError("MissingRevision", "git sources are expected to specify a Revision, got: %v", git)
 	}
+
+	args := []string{"-url", git.Url,
+		"-revision", git.Revision,
+	}
+	containerName := initContainerPrefix + gitSource
+
+	if name != "" {
+		// update container name to suffix source name
+		args = append(args, []string{"-name", name}...)
+		containerName = containerName + "-" + name
+	}
+
 	return &corev1.Container{
-		Name:  initContainerPrefix + gitSource,
-		Image: *gitImage,
-		Args: []string{
-			"-url", git.Url,
-			"-revision", git.Revision,
-		},
+		Name:         containerName,
+		Image:        *gitImage,
+		Args:         args,
 		VolumeMounts: implicitVolumeMounts,
 		WorkingDir:   workspaceDir,
 		Env:          implicitEnvVars,
 	}, nil
 }
 
 func containerToGit(git corev1.Container) (*v1alpha1.SourceSpec, error) {
+	// git args are expected to be in this order "--url url --revision revision (optional)--name name ""
+
 	if git.Image != *gitImage {
 		return nil, fmt.Errorf("Unrecognized git source image: %v", git.Image)
 	}
 	if len(git.Args) < 3 {
 		return nil, fmt.Errorf("Unexpectedly few arguments to git source container: %v", git.Args)
 	}
+
+	name := ""
+
+	// if `--name name` is set in args extract $name variable
+	if len(git.Args) == 6 {
+		name = git.Args[5]
+	}
+
 	// Now undo what we did above
 	return &v1alpha1.SourceSpec{
+		Name: name,
 		Git: &v1alpha1.GitSourceSpec{
 			Url:      git.Args[1],
 			Revision: git.Args[3],
 		},
 	}, nil
 }
 
-func gcsToContainer(gcs *v1alpha1.GCSSourceSpec) (*corev1.Container, error) {
+func gcsToContainer(gcs *v1alpha1.GCSSourceSpec, name string) (*corev1.Container, error) {
 	if gcs.Location == "" {
 		return nil, validation.NewError("MissingLocation", "gcs sources are expected to specify a Location, got: %v", gcs)
 	}
+	// source name is empty then use `build-step-gcs-source` name
+	containerName := initContainerPrefix + gcsSource
+
+	if name != "" {
```

Comment:
Also add the `--name` flag?

Reply:
README does not document `dest_dir`. I looked into codebase to get that flag. I will update it here and e2e test. Thanks fir catching that.